README.md

```diff
@@ -1,3 +1,134 @@
+to generate the optimal state variable values.
+
+![fuse graph](doc/fuse_graph_4.png)
+
+While our `fuse` system is now optimizing constraints from two different sensors and a motion model, it is not yet
+publishing any data back out to ROS. We should probably do that. So we would derive a `fuse_core::Publisher` class
+that finds the most recent position and orientation and publishes that as a `geometry_msgs::PoseStamped` message to
+the topic of your choice.
+
+![fuse graph](doc/fuse_graph_5.png)
+
+We finally have something that is starting to be useful.
+
+### Adaptation #1: Full path publishing
+
+Nothing about the `fuse` framework limits you to having a single publisher. What if you want to visualize the entire
+robot trajectory, instead of just the most recent pose? Well, we can create a new derived `fuse_core::Publisher` class
+that publishes all of the robot poses using a `nav_msgs::Path` message.
+
+![fuse graph](doc/fuse_graph_adaptation1.png)
+
+### Adaptation #2: Changing kinematics
+
+In your spare time, you also build [autonomous power wheels racers](http://www.powerracingseries.org/). But race cars
+don't use differential drive; you need a different motion model. Easy enough. We simply derive a new
+`fuse_core::MotionModel` class that implements an Ackermann steering model. Everything else can be reused.
+
+![fuse graph](doc/fuse_graph_adaptation2.png)
+
+### Adaptation #3: Online calibration
+
+Over time you notice that the accuracy of the odometry measurements is decreasing. After some investigation you realize
+that the soft rubber racing tires are wearing, decreasing the diameter of the wheels over time. It sure would be nice
+if the odometry system could compensate for that automatically. To do that, we will create a new variable type, a wheel
+diameter. We will also need a new wheel encoder sensor model. This new model will involve the previous pose and
+next pose as before, but it will also involve the previous wheel diameter estimate. Finally, we will need to create
+a "motion model" that describes how the wheel diameter is expected to change over time. Maybe some sort of exponential
+decay?
+
+![fuse graph](doc/fuse_graph_adaptation3.png)
+
+Now our system estimates the wheel diameters at each time step as well as the robot's pose.
+
+## Summary
+The purpose of `fuse` is to provide a framework for performing sensor fusion tasks, allowing common components to be
+reused between systems, while also allowing components to be customized for different use cases. Since all of the
+models are implemented as plugins, separate plugin libraries can be shared or kept private at the discretion of their
+authors without have to fork the `fuse` repository.
+
+## The Math
+I was going to write a nice nonlinear least squares inference summary, but I honestly couldn't do any better than the
+one provided by the Google Ceres [project](http://ceres-solver.org/nnls_modeling.html).
```

Comment:
This is the first mention of Ceres. I think that you are underselling things a bit. I would put a big note up at the top saying that the core `fuse` is reconciling multiple different conflicting sensors by finding a best fit using the beefy modelling of Ceres, or something to that effect.